src/pymor/operators/block.py

```diff
@@ -131,6 +131,12 @@ def d_mu(self, parameter, index=0):
             blocks[i, j] = self.blocks[i, j].d_mu(parameter, index)
         return self.with_(blocks=blocks)
 
+    def jacobian(self, U, mu):
+        jacs = np.empty(self.blocks.shape, dtype=object)
+        for (i, j) in np.ndindex(self.blocks.shape):
+            jacs[i, j] = self.blocks[i, j].jacobian(U if self.num_source_blocks == 1 else U.blocks[i], mu)
```

Comment:
If I'm not mistaken, you should check `self.blocked_source` instead of `num_source_blocks`. It still can happen that `U` is a `BlockVectorArray` with only one block. I don't think we have any magic in place that would allow using `U` instead of `U.blocks[0]` in that case.